tests/datalayer.unit.test

```diff
@@ -0,0 +1,389 @@
+    $item_static = $item;
+    $return_type = FALSE;
+    $result = _datalayer_menu_get_any_object($return_type);
+    $this->assertEqual($return_type, FALSE);
+    $this->assertEqual($result, FALSE);
+  }
+
+  /**
+   * Test DataLayer Menu Get Any Object.
+   *
+   * Returns False Without Load Function Match.
+   */
+  public function testDataLayerMenuGetAnyObjectReturnsFalseWithoutLoadFunctionMatch() {
+    $item = $this->setupMockNode();
+    $item['node/1']['load_functions'] = array(1 => 'user_load');
+    $item_static = &drupal_static('menu_get_item');
+    $item_static = $item;
+    $return_type = FALSE;
+    $result = _datalayer_menu_get_any_object($return_type);
+    $this->assertEqual($return_type, FALSE);
+    $this->assertEqual($result, FALSE);
+  }
+
+  /**
+   * Test DataLayer Menu Get Any Object.
+   *
+   * Returns False With Incorrect Arg Position.
+   */
+  public function testDataLayerMenuGetAnyObjectReturnsFalseWithIncorrectArgPosition() {
+    $item = $this->setupMockNode();
+    $item['node/1']['load_functions'] = array('user_load');
+    $item_static = &drupal_static('menu_get_item');
+    $item_static = $item;
+    $return_type = FALSE;
+    $result = _datalayer_menu_get_any_object($return_type);
+    $this->assertEqual($return_type, FALSE);
+    $this->assertEqual($result, FALSE);
+  }
+
+  /**
+   * Test DataLayer Menu Get Any Object Returns Object.
+   */
+  public function testDataLayerMenuGetAnyObjectReturnsObject() {
+    $item = $this->setupMockNode();
+    $return_type = FALSE;
+    $object = _datalayer_menu_get_any_object($return_type);
+    $this->assertEqual($return_type, 'node');
+    $this->assertEqual($object, $item['node/1']['map'][1]);
+  }
+
+  /**
+   * Test DataLayer Get Entity Terms Returns Empty Array.
+   */
+  public function testDataLayerGetEntityTermsReturnsEmptyArray() {
+    $item = $this->setupMockNode();
+    $this->setupMockFieldMap();
+    $terms = _datalayer_get_entity_terms('node', 'page', $item['node/1']['map'][1]);
+    $this->assertEqual(array(), $terms);
+  }
+
+  /**
+   * Test DataLayer Get Entity Terms Returns Term Array.
+   */
+  public function testDataLayerGetEntityTermsReturnsTermArray() {
+    $item = $this->setupMockNode();
+    $this->setupMockEntityTerms();
+    $terms = _datalayer_get_entity_terms('node', 'article', $item['node/1']['map'][1]);
+    $this->assertEqual(array('tags' => array(1 => 'someTag')), $terms);
+  }
+
+  /**
+   * Test DataLayer Get Entity Terms Returns Entity Data Array.
+   */
+  public function testDataLayerGetEntityDataReturnsEntityDataArray() {
+    $this->setupEmptyDataLayer();
+    $item = $this->setupMockNode();
+    $this->setupMockEntityTerms();
+    $entity_data = _datalayer_get_entity_data($item['node/1']['map'][1], 'node');
+    $this->assertEqual(
+      $this->getExpectedEntityDataArray(),
+      $entity_data
+    );
+  }
+
+  /**
+   * Setup user.
+   */
+  public function setupMockUser() {
```

Comment:
Good call.

I think I started it to help test `datalayer_preprocess_page` but then I got lazy and submitted the patch as "pass one."

I'm going to leave it for "pass two."